File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0];
        int ans = 0;

        for(int val : prices){
            if(val < min)min=val;
            else ans = ans > val - min ? ans : val - min;
        }

        return ans;

    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice=2**31 - 1
        maxProfit=0
        for price in prices:
            if price < minPrice : minPrice=price
            else :
                # print(price)
                maxProfit=max(maxProfit,price-minPrice)

        return maxProfit        